sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -625,43 +760,50 @@ export class EventHubReceiver extends LinkEntity {
       }
     } catch (err) {
       this.isConnecting = false;
-      err = translate(err);
+      const error = translate(err);
       log.error(
         "[%s] An error occured while creating the receiver '%s': %O",
         this._context.connectionId,
         this.name,
-        err
+        error
       );
-      throw err;
+      throw error;
     }
   }
 
   /**
    * Creates the options that need to be specified while creating an AMQP receiver link.
    * @ignore
    */
-  protected _createReceiverOptions(options: CreateReceiverOptions): RheaReceiverOptions {
-    if (options.newName) this.name = `${uuid()}`;
+  private _createReceiverOptions(options: CreateReceiverOptions): RheaReceiverOptions {
+    if (options.newName) this.name = uuid();
     const rcvrOptions: RheaReceiverOptions = {
       name: this.name,
       autoaccept: true,
       source: {
         address: this.address
       },
       credit_window: this.prefetchCount,
-      onMessage: options.onMessage || this._onAmqpMessage,
-      onError: options.onError || this._onAmqpError,
-      onClose: options.onClose || this._onAmqpClose,
-      onSessionError: options.onSessionError || this._onSessionError,
-      onSessionClose: options.onSessionClose || this._onSessionClose
+      onMessage: options.onMessage || ((context: EventContext) => this._onAmqpMessage(context)),
+      onError: options.onError || ((context: EventContext) => this._onAmqpError(context)),
+      onClose: options.onClose || ((context: EventContext) => this._onAmqpClose(context)),
+      onSessionError:
+        options.onSessionError || ((context: EventContext) => this._onAmqpSessionError(context)),
+      onSessionClose:
+        options.onSessionClose || ((context: EventContext) => this._onAmqpSessionClose(context))
     };
-    if (this.ownerLevel !== undefined && this.ownerLevel !== null) {
-      if (!rcvrOptions.properties) rcvrOptions.properties = {};
-      rcvrOptions.properties[Constants.attachEpoch] = types.wrap_long(this.ownerLevel);
+
+    if (typeof this.ownerLevel === "number") {
+      if (!rcvrOptions.properties)
```

Comment:
`rcvrOptions` is being created only a few lines before, and we know it doesnt have `properties`, so why this check?

Reply:
That's a good point... admittedly there from when I copy pasted the existing code but I will remove the check.